src/components/Todo/Todo.js

```diff
@@ -1,56 +1,128 @@
 import React, { Component } from 'react' ;
 import {TextField  , Button , List} from '@material-ui/core'
 import classes from './Todo.module.css';
+import axios from '../../axios-ListData';
 
 import ListTodo from './ListTodo/ListTodo'
+import withErrorHandler from '../../hoc/withErrorHandler/withErrorHandler';
 
 class Todo extends Component {
 
     state = {
-        value : '' ,
-        ListTodo : {
-          list1: {  
-            todoText : 'ahmad', 
-            checked : true ,
-          }
-        }
+       todoList : [] ,
+       value : '' ,
+       id : '',
+       checked : false,
+       
     }
 
+    componentDidMount () {
+      axios.get('/ListTodo.json')
+            .then(response => {    
+              let updateTodoData = [] ;
+              let todoData = Object.keys(response.data).map(item => {
+                return {...response.data[item]  ,id : item}  })
+               for( let key in todoData) {
+                  if(todoData[key].Date === new Date().toDateString()){
+                         console.log(todoData[key].Date)
+                         updateTodoData.push(todoData[key])
+                  }
+
+                }
+                  console.log(updateTodoData)
+              this.setState({
+                  todoList : updateTodoData
+              })
+            }).catch(err => {
+              console.log(err)
+            })
+    }
+
+
     handleChange = (event) => {
-       event.preventDefault();
-      
        this.setState({
              value : event.target.value,
        })
+
+    }
+
+    onAddTodoHandler = (event , inputValue) => {
+      event.preventDefault();
+      const newArr = [...this.state.todoList];
+      const todoData = {
+        name : inputValue ,
+        Date : new Date().toDateString(),
+        checked : false ,
+      }
+      axios.post("/ListTodo.json" , todoData)
+            .then(response => {
+              newArr.push({...todoData ,id:response.data.name})
+              this.setState({
+                todoList : newArr ,
+                value : ''
+              })
+            }).catch(err => {
+              console.log(err)
+            })
     }
+    onRemoveTodoTasks = index => {
+        axios.delete(`/ListTodo/${index}.json`)
+              .then(res => {
+                this.setState( previousState => {
+                  return {
+                    todoList : previousState.todoList.filter(item => item.id !== index)
+                  };
+                });
+              }).catch(err=>{
+                console.log(err);
+              })
+       
+    }
+
+    onCheckBtnHandler = (index , item) => {
+      const updateChecked = {
+          ...item,
+          checked : !item.checked
+      }
+      axios.put(`/ListTodo/${item.id}.json`,updateChecked)
+             .then(res => {
+               let newArr = [...this.state.todoList.filter(item => index !== item.id)];
```

Comment:
This is not wrong, but please be aware that some State is immutable, and we should not update it directly. Some funtions do modification on item (filter doesn't update the list as it returns a new array) but take this in considrations